impeller/docs/standalone_gles.md

```diff
@@ -0,0 +1,195 @@
+# Using Impeller as a Standalone Rendering Library (with OpenGL ES)
+
+This guide describes how to use Impeller as a standalone rendering library using OpenGL ES. Additionally, some form of WSI (Window System Integration) is essential. Since EGL is the most the most popular form of performing WSI on platforms with OpenGL ES, Impeller has a toolkit that assists in working with EGL. This guide will use that toolkit.
+
+While this guide focuses on OpenGL ES with EGL, the steps to setup rendering with another client rendering API (Metal and Vulkan) are fairly similar and you should be able to follow the same pattern for other backends.
+
+This guide details extremely low-level setup and the `//impeller/renderer` API directly above the HAL. Most users of Impeller will likely use the API using convenience wrappers already written for the platform. Interacting directly with the HAL is extremely powerful but also verbose. Applications are likely to also use higher level frameworks like Aiks or Display Lists.
+
+Building Impeller for the target platform is outside the scope of this guide.
+
+> [!CAUTION]
+> The code provided inline is pseudo-code and doesn't include error handling. See the headerdocs for more on error handling and failure modes. All classes are assumed to be in the `impeller` namespace. For a more complete example of setting up standalone Impeller, see [this patch](https://github.com/flutter/engine/pull/52472/files) that adds support for Impeller rendering via Wasm [in the browser and WebGL 2](https://public.chinmaygarde.com/impeller/wasm/wasm.html).
+
+# Setup
+
+To get started with Impeller rendering, you need to setup a context and a renderer. For backend specific classes, the convention in Impeller is to append the backend name as a suffix. So if you need to create an instance of an `impeller::Context` for OpenGLES, look for `impeller::ContextGLES`.
+
+In our case, we need a `impeller::ContextGLES` and `impeller::Renderer`.
+
+## Setting up WSI with the EGL Toolkit
+
+Before we get to `OpenGLES` we need to do a bit of WSI. While we can resort to using EGL directly, Impeller has a handy toolkit for it.
+
+First create an EGL display connection:
+
+```c++
+egl::Display display;
```

Comment:
nit: would be nice to include headers for the various symbols referenced, although that would make this doc a bit more fragile.

Reply:
Yeah, I figured linking the patch would be sufficient and leaving the includes for brevity.

Reply:
fwiw (not saying we need to do this here), what we do in the flutter/flutter repo is literally extract out the \`\`\` bits of code and statically analyze them, and what we do in the flutter/packages repo is have the README files explicitly link to actual code that is tested and then enforce that the README file snippets match the actual code. In both cases the result is that code in documentation is checked on every commit.

Reply:
I spoke to @yjbanov and I may just check in the code. Previously, I didn't want to setup the toolchain and testing for a target just for documentation.